src/lib/optimizer/strategy/predicate_merge_rule.cpp

```diff
@@ -0,0 +1,127 @@
+#include "predicate_merge_rule.hpp"
+
+#include "expression/expression_functional.hpp"
+#include "expression/logical_expression.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/union_node.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+void PredicateMergeRule::apply_to(const std::shared_ptr<AbstractLQPNode>& root) const {
+  Assert(root->type == LQPNodeType::Root, "PredicateMergeRule needs root to hold onto");
+
+  // For every node, store a pointer to the root of the subplan if this subplan can be merged
```

Comment:
Good point. But the algorithm works differently: Instead of looking for disjunctive chains it counts the total number of `UnionNodes` (with `UnionMode::Positions`) in each subplan that only consists of `UnionNodes` (with `UnionMode::Positions`) and `PredicateNodes`. I changed the variable names accordingly.